src/layouts/main/styles/functions.js

```diff
@@ -0,0 +1,74 @@
+import {css} from 'styled-components';
+import {drawerWidth} from './constants';
+
+/**
+ * To return drawer styles when is open.
+ *
+ * @param {object} theme
+ * @return {css} drawer styles when opened
+ */
+const getOpenDrawerStyles = (theme) => css`
+  width: ${drawerWidth}px;
+  transition: ${theme.transitions.create('width', {
+    easing: theme.transitions.easing.sharp,
+    duration: theme.transitions.duration.enteringScreen,
+  })};
+`;
+
+/**
+ * To return drawer styles when is closed.
+ *
+ * @param {object} theme
+ * @return {css} drawer styles when closed
+ */
+const getClosedDrawerStyles = (theme) => css`
+  transition: ${theme.transitions.create('width', {
+    easing: theme.transitions.easing.sharp,
+    duration: theme.transitions.duration.leavingScreen,
+  })};
+  overflow-x: hidden;
+  width: ${theme.spacing(9) + 1}px;
+
+  .MuiListItemText-root {
+    visibility: hidden;
+  }
+`;
+
+/**
+ * To return drawer styles according to
+ * the opened state
+ *
+ * @param {{open: boolean, theme: object}} props
+ * @return {css} styles according to open state
+ */
+const drawerToggles = ({open, theme}) => {
+  if (open) return getOpenDrawerStyles(theme);
+  return getClosedDrawerStyles(theme);
+};
+
+/**
+ * To return menu tab styles when is active.
+ *
+ * @param {object} theme
+ * @return {css} menu tab styles when active
+ */
+const getActiveTabStyles = (theme) => css`
+  color: ${theme.palette.primary.main};
+
+  .MuiSvgIcon-root {
+    color: ${theme.palette.primary.main};
+  }
+`;
+
+/**
+ * To return active styles for navigation tab
+ * only when is active
+ *
+ * @param {{active: boolean, theme: object}} props
+ * @return {css} styles according to ac state
+ */
+const tabToggle = ({active, theme}) => {
```

Comment:
I'm not sure I understand the function naming for this? Also, would it be possible to merge this function with getActiveTabStyles?

Reply:
Some context: I had some problems with the css-in-js, since it uses indentation inside JS template literals, and the linter shows an indentation error. That's why I decided to put these styles in subfunctions since it respects the indentation. I'm attaching an SS for that:

![ESLint Error](https://i.ibb.co/PTd7xTf/Captura-de-Pantalla-2020-08-03-a-la-s-0-24-50.png)
![Original function](https://i.ibb.co/mCkcFX2/Captura-de-Pantalla-2020-08-03-a-la-s-0-26-10.png)

That been said, I tried to use only `getActiveTabStyles` as suggested, and it worked, so I replaced it 😄 , I'm just saying this just to show that we might have to add a special rule in the ESLint configuration to avoid indentation errors. In that case, I can simplify the way we use these functions, which is important since it'll be frequent to have them.